js/src/builder/valid.ts

```diff
@@ -0,0 +1,64 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import { DataType } from '../type';
+import { valueToString } from '../util/pretty';
+import { BigIntAvailable } from '../util/compat';
+
+/**
+ * Dynamically compile the null values into an `isValid()` function whose
+ * implementation is a switch statement. Microbenchmarks in v8 indicate
+ * this approach is 25% faster than using an ES6 Map.
+ * @ignore
+ * @param nullValues
+ */
```

Comment:
Could you add an example of what the generated function should look like here? I find that pretty helpful when looking through code and encountering some kind of code generation.